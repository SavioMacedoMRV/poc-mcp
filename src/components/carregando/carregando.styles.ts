import Lottie from 'lottie-react'
import styled from 'styled-components'
import animacao from 'animations/carregando.json'

export const Container = styled.div`
  display: flex;
  flex: 1;
  height: 100vh;
  align-items: center;
  justify-content: center;
`
export const Animacao = styled(Lottie).attrs({
  animationData: animacao,
})`
  width: 350px;
`
