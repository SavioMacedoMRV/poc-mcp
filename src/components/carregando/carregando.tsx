import React from 'react'
import animacao from 'animations/carregando.json'
import * as S from './carregando.styles'

export function Carregando() {
  return (
    <S.Container>
      <S.Animacao data-testid="carregando" animationData={animacao} />
    </S.Container>
  )
}
