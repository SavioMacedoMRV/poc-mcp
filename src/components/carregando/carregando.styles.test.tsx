import {render} from '@testing-library/react'
import * as S from './carregando.styles'

describe('Carregando.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Container />)
    expect(container).toMatchSnapshot()
  })

  test('Animacao DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Animacao animationData={{json: 'json'}} />)

    expect(container).toMatchSnapshot()
  })
})
