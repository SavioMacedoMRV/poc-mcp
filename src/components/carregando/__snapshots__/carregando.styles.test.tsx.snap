// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carregando.styles Animacao DEVE ser igual ao snapshot 1`] = `
.c0 {
  width: 350px;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Carregando.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  flex: 1;
  height: 100vh;
  align-items: center;
  justify-content: center;
}

<div>
  <div
    class="c0"
  />
</div>
`;
